### PR TITLE
Ftr: add dubbogo3 demo

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
Add  grpc and dubbo3 demo with same IDL. 
You can try any server with any client to test their connection.

### CODE REVIEW

File: golang/go.mod [hunk 1]
```diff
@@ -4,17 +4,20 @@ require (
...
 	github.com/prometheus/client_golang v1.1.0
 	github.com/uber/jaeger-client-go v2.22.1+incompatible
 	github.com/uber/jaeger-lib v2.2.0+incompatible // indirect
+	golang.org/x/net v0.0.0-20200822124328-c89045814202
 	google.golang.org/grpc v1.26.0
 )
 
 go 1.13
+
+replace github.com/apache/dubbo-go v1.5.2-0.20200919130513-8a059ddd7bf7 => ../../dubbo-go
```
Comment (LaurenceLiZhixin) [inline golang/go.mod:23]: I 'll delete this after my dubbo3 support merged to apach/dubbo-go 3.0

File: golang/general/dubbo3/server/dubbo3-server/cmd/server.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"github.com/apache/dubbo-samples/golang/general/dubbo3/server/dubbo3-server/pkg"
```
Comment (IronCity) [inline golang/general/dubbo3/server/dubbo3-server/cmd/server.go:22]: split

Comment (cityiron) [inline golang/general/dubbo3/server/dubbo3-server/cmd/server.go:22]: split

File: golang/general/dubbo3/server/dubbo3-server/pkg/greeter.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
+
+import (
+	"context"
+	"fmt"
+	"github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/dubbo3"
```
Comment (cityiron) [inline golang/general/dubbo3/server/dubbo3-server/pkg/greeter.go:23]: split

File: golang/general/dubbo3/server/dubbo3-server/pkg/greeter.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+type GreeterProvider struct {
```
Comment (cityiron) [inline golang/general/dubbo3/server/dubbo3-server/pkg/greeter.go:26]: miss comment

File: golang/general/dubbo3/server/server/main.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	pb "github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/grpc"
```
Comment (cityiron) [inline golang/general/dubbo3/server/server/main.go:22]: split

File: golang/general/dubbo3/unary-client/grpc-client/main.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	pb "github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/grpc"
```
Comment (cityiron) [inline golang/general/dubbo3/unary-client/grpc-client/main.go:21]: split

File: golang/general/dubbo3/unary-client/dubbo3-client/pkg/hello.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
+	"context"
+	_ "github.com/apache/dubbo-go/protocol/dubbo3/impl"
```
Comment (cityiron) [inline golang/general/dubbo3/unary-client/dubbo3-client/pkg/hello.go:22]: split

File: golang/general/dubbo3/unary-client/dubbo3-client/cmd/client.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/dubbo3"
```
Comment (cityiron) [inline golang/general/dubbo3/unary-client/dubbo3-client/cmd/client.go:22]: split

File: golang/general/dubbo3/stream-client/grpc-client/main.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	pb "github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/grpc"
```
Comment (cityiron) [inline golang/general/dubbo3/stream-client/grpc-client/main.go:22]: split

File: golang/general/dubbo3/stream-client/dubbo3-client/pkg/hello.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pkg
+
+import (
+	"context"
+	_ "github.com/apache/dubbo-go/protocol/dubbo3/impl"
```
Comment (cityiron) [inline golang/general/dubbo3/stream-client/dubbo3-client/pkg/hello.go:22]: split

File: golang/general/dubbo3/stream-client/dubbo3-client/cmd/client.go [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/apache/dubbo-samples/golang/general/dubbo3/protobuf/dubbo3"
```
Comment (cityiron) [inline golang/general/dubbo3/stream-client/dubbo3-client/cmd/client.go:23]: split

**Pull request closed without merging**